### ISSUE TITLE
Panorama XML reloading bugs out with panels added with BLoadLayoutSnippet

### ISSUE THREAD

**Issue by DoctorGester** [ModDota/BugTracker#103]
As the title states. Things never get cleaned up unless you delete a panel manually and readd/RemoveAndDeleteChildren() on init it and create a mess.

![image](https://cloud.githubusercontent.com/assets/3684394/23282565/d3f77676-fa32-11e6-860f-d1e61892d302.png)

**Comment by Arhowk** [comment]
That is correct. Panorama reloading is not a full reload of the system but rather a difference check-style system. This same interaction can be found with CSS classes, if you apply a CSS class to a panel, load it, remove it, reload it, the panel will still have the same CSS class.

**Comment by DoctorGester** [comment]
These are both annoying bugs.

**Comment by Arhowk** [comment]
$('#yourPanel').RemoveAndDeleteChildren() on JS run.

**Comment by DoctorGester** [comment]
This is not a solution.

**Comment by Arhowk** [comment]
Elaborate?

**Comment by DoctorGester** [comment]
Really? Do you realize I can be adding panels to elements which ALREADY contain panels from XML? If you have read the original post you would have seen that I already mentioned RemoveAndDeleteChildren. This is NOT a correct behavior.

**Comment by DoctorGester** [comment]
```xml
<root>
    <snippets>
        <snippet name="TestBoob">
            <Panel>
                <Panel style="width:100px; height: 16px; background-color: red;"></Panel>
            </Panel>
        </snippet>
    </snippets>
    <Panel style="width: 100%; height: 100%;">
        <Panel id="TestPanel" style="flow-children: down; background-color: white;"/>
        <script>
            $("#TestPanel").BLoadLayoutSnippet("TestBoob");
        </script>
    </Panel>
</root>
```

The code in question creates TWO snippet panels for me.

Changing the script to:

```xml
<script>
            var pp = $.CreatePanel("Panel", $("#TestPanel"), "");
            pp.BLoadLayoutSnippet("TestBoob");
        </script>
```

Adds an extra panel on each reload.

**Comment by DoctorGester** [comment]
Also when I use the code number 1 from the previous comment the TestPanel completely bugs out. It's visible even if I delete it and the code from the XML. If I also delete the snippet the empty panel is left. My guess would be that panorama fails do delete snippet parents during reload.
